### DISCUSSION TITLE
Allow wildcard options to clearing localstorage

### DISCUSSION THREAD

**Discussion by ryanbr** [uBlockOrigin/uAssets#20194] [Ideas]
We had a recent issue with [crunchyroll.com/evidon and GPC](https://github.com/brave/brave-browser/issues/31954).

We block GPC by default (also DDG, https://github.com/duckduckgo/privacy-configuration/issues/1422 and Firefox Nightly/Enhanced Protection)

We implemented a fix to prevent _evidon_consent_ls local item being set by blocking ||c.evidon.com/dg/dg.js$script https://github.com/easylist/easylist/commit/2633080da1b37b92d364f70cdb32c814f439eb6f

The issue currently is, if this local stoage item `_evidon_consent_ls_[randomchars]` is still set for exisiting users, it will still cause GPC login issues until the user manually clears this specific stoage item.

If possible to allow to remove a name based on wildcard (or regex). Given its only clearing data, doesn't look like a privacy issue?

Something like this:
`sso.crunchyroll.com##+js(set-local-storage-item, _evidon_consent_ls_*, $remove$)`

or via regex
`crunchyroll.com##+js(set-local-storage-item, /_evidon_consent_ls_\d+/, $remove$)`

As an interim; https://github.com/brave/adblock-resources/blob/test-delete-local-storage-script/resources/brave-delete-from-local-storage.js

**Comment by gorhill** [comment]
https://github.com/gorhill/uBlock/commit/41d91ed0809db8fe2208b7811c9dc7d11f06f889
